src/data_provider/src/packages/packagesWindows.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Wazuh SYSINFO
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ * January 24, 2022.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _PACKAGE_WINDOWS_H
+#define _PACKAGE_WINDOWS_H
+
+#include <windows.h>
+#include "ipackageInterface.h"
+#include "ipackageWrapper.h"
+
+class FactoryWindowsPackage
+{
+    public:
+        static std::shared_ptr<IPackage>create(const HKEY key, const std::string& subKey);
+};
+
+class WindowsPackageImpl final : public IPackage
+{
+        const std::shared_ptr<IPackageWrapper> m_packageWrapper;
```

Comment:
Despite this class variable being private without explicitly declaring it as private, I think adding it will be helpful ;)